### PR TITLE
indexer easy: a couple of clean-ups

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

- chatted w/ Patrick and we agreed to remove in-memory indexer store, b/c we have PG services on GH. Also it slows ppl down a bit as it requires an additional change each time things happen on IndexerStore.
- removed unnecessary partitions b/c partition manager is in place and can proceed

## Test Plan 

- CI

### CODE REVIEW

**Review by 666lcz — APPROVED** [review]

File: crates/sui-indexer/migrations/2022-12-01-034426_objects/up.sql [hunk 1]
```diff
@@ -48,13 +48,7 @@ CREATE TABLE objects_history
     bcs                    bcs_bytes[]   NOT NULL,
     CONSTRAINT objects_history_pk PRIMARY KEY (epoch, object_id, version)
 ) PARTITION BY RANGE (epoch);
--- TODO: Add trigger to automatically create partitions at new epoch when we have Epoch Table
 CREATE TABLE objects_history_partition_0 PARTITION OF objects_history FOR VALUES FROM (0) TO (1);
```
Comment (wlmyng) [inline crates/sui-indexer/migrations/2022-12-01-034426_objects/up.sql:51]: we dont need to remove this one?

**Review by wlmyng — APPROVED** [review]
:ok:

File: crates/sui-indexer/migrations/2022-12-01-034426_objects/up.sql (hunk 1 above)
Comment (gegaowp) [reply crates/sui-indexer/migrations/2022-12-01-034426_objects/up.sql:51]: good q, we actually need to keep this b/c of the advance logic here
https://github.com/MystenLabs/sui/blob/main/crates/sui-indexer/src/store/pg_indexer_store.rs#L953-L956

Comment (wlmyng) [reply crates/sui-indexer/migrations/2022-12-01-034426_objects/up.sql:51]: oh okay I may have misconstrued/ confused a few things. This is not the in-memory indexer store, this is the pg indexer store. And we need to instantiate one, the pg indexer will then create tables as needed off this one

Comment (gegaowp) [reply crates/sui-indexer/migrations/2022-12-01-034426_objects/up.sql:51]: this is for PG indeed. this is for table partitioning, for object_history for example, the reason of having partitioning is to optimize the query perf by limiting size of each partition. 
when we query, we still query the "wrapper table" or "object_history" and do not have to worry about underlying partitions like "object_history_partition_0"

**Pull request merged**